File: dataset_reduzido/codenet/Java/Code/s821112618.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] s = ".,!? :abc:def:ghi:jkl:mno:pqrs:tuv:wxyz".split(":");
        char[][] k = new char[9][];
        for (int i = 0; i < 9; i++) {
            k[i] = s[i].toCharArray();
        }
        while (n-- > 0) {
            char[] c = sc.next().toCharArray();
            int l = -1, a = 0;
            for (int i = 0; i < c.length; i++) {
                if (c[i] != '0') {
                    a = c[i] - '1';
                    l = (l + 1) % k[a].length;
                } else if (l >= 0) {
                    System.out.print(k[a][l]);
                    l = -1;
                }
            }
            System.out.println();
        }
    }
}

File: output/StarCoder/codenet/Java/Python/s821112618.py
def main():
    n = int(input())
    s = ".,!? :abc:def:ghi:jkl:mno:pqrs:tuv:wxyz".split(":")
    k = [list(char) for char in s[:-1]]
    for _ in range(n):
        c = input()
        l = -1
        for i in range(len(c)):
            if c[i] != '0':
                a = int(c[i]) - 1
                l = (l + 1) % len(k[a])
            elif l >= 0:
                print(k[a][l], end='')
                l = -1
        print()

if __name__ == "__main__":
    main()